src/components/ConnectToSageIntacctButton/index.tsx

```diff
@@ -0,0 +1,138 @@
+import Navigation from '@libs/Navigation/Navigation';
+import {getPoliciesConnectedToSageIntacct} from '@libs/PolicyUtils';
+import type {AnchorPosition} from '@styles/index';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import type {PolicyConnectionName} from '@src/types/onyx/Policy';
+
+type ConnectToSageIntacctButtonProps = {
+    policyID: string;
+    shouldDisconnectIntegrationBeforeConnecting?: boolean;
+    integrationToDisconnect?: PolicyConnectionName;
+};
+
+function ConnectToSageIntacctButton({policyID, shouldDisconnectIntegrationBeforeConnecting, integrationToDisconnect}: ConnectToSageIntacctButtonProps) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    const {isOffline} = useNetwork();
+    const currentIntegration = translate('workspace.accounting.integrationName', integrationToDisconnect);
+    const integrationToConnect = translate('workspace.accounting.integrationName', CONST.POLICY.CONNECTIONS.NAME.SAGE_INTACCT);
+
+    const [isDisconnectModalOpen, setIsDisconnectModalOpen] = useState(false);
+
+    const [policies] = useOnyx(ONYXKEYS.COLLECTION.POLICY);
+    const hasPoliciesConnectedToSageIntacct = !!getPoliciesConnectedToSageIntacct(policies).length;
+    const {isSmallScreenWidth} = useWindowDimensions();
+    const [isReuseConnectionsPopoverOpen, setIsReuseConnectionsPopoverOpen] = useState(false);
+    const [reuseConnectionPopoverPosition, setReuseConnectionPopoverPosition] = useState<AnchorPosition>({horizontal: 0, vertical: 0});
+    const threeDotsMenuContainerRef = useRef<View>(null);
+    const connectionOptions = [
+        {
+            icon: Expensicons.LinkCopy,
+            text: translate('workspace.intacct.createNewConnection'),
+            onSelected: () => {
+                Navigation.navigate(ROUTES.POLICY_ACCOUNTING_SAGE_INTACCT_PREREQUISITES.getRoute(policyID));
+                setIsReuseConnectionsPopoverOpen(false);
+            },
+        },
+        {
+            icon: Expensicons.Copy,
+            text: translate('workspace.intacct.reuseExitingConnection'),
+            onSelected: () => {
+                Navigation.navigate(ROUTES.POLICY_ACCOUNTING_SAGE_INTACCT_EXISTING_CONNECTIONS.getRoute(policyID));
+                setIsReuseConnectionsPopoverOpen(false);
+            },
+        },
+    ];
+
+    return (
+        <>
+            <Button
+                onPress={() => {
+                    if (shouldDisconnectIntegrationBeforeConnecting && integrationToDisconnect) {
+                        setIsDisconnectModalOpen(true);
+                        return;
+                    }
+                    if (!hasPoliciesConnectedToSageIntacct) {
+                        Navigation.navigate(ROUTES.POLICY_ACCOUNTING_SAGE_INTACCT_PREREQUISITES.getRoute(policyID));
+                        return;
+                    }
+                    if (!isSmallScreenWidth) {
```

Comment:
I am a little confused on why do we need this?

Reply:
When we have big screen, instead of drawer 
<img width="264" alt="Screenshot 2024-06-27 at 10 57 26" src="https://github.com/Expensify/App/assets/88395093/579103f4-bbaa-4cf5-a041-b75193e9031a">
there is a pop up:
<img width="851" alt="Screenshot 2024-06-27 at 10 57 41" src="https://github.com/Expensify/App/assets/88395093/6ba4cdc0-5251-4f25-87c0-3d6290575a8c">
and that's why we need to set these x and y coords